httpx/middleware/newrelic_tracer_test.go

```diff
@@ -0,0 +1,142 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/remind101/nra"
+	"github.com/remind101/pkg/httpx"
+	"github.com/stretchr/testify/mock"
+	"golang.org/x/net/context"
+)
+
+type tracerTest struct {
+	// A function that adds Handlers to the router.
+	routes func(*httpx.Router)
+
+	// An http.Request to test.
+	req *http.Request
+
+	expectedTransactionName string
+	expectedUrl             string
+}
+
+func TestTracing(t *testing.T) {
+	tracerTests := []tracerTest{
+		// simple path
+		{
+			routes: func(r *httpx.Router) {
+				r.Handle("/path", httpx.HandlerFunc(func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+					return nil
+				})).Methods("GET")
+			},
+			req: newRequest("GET", "/path"),
+			expectedTransactionName: "/path (GET)",
+			expectedUrl:             "/path",
+		},
+		// path with variables
```

Comment:
Just for sanity, might wanna add a test with a regex param like `/articles/{category}/{id:[0-9]+}`.